### PR TITLE
Refactor SpectrumDataset to use exposure

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request refactors the `SpectrumDataset` as well as `SpectrumDatasetOnOff` to use exposure internally as well. To support round-tripping the OGIP format in introduces the `livetime` meta keyword in the exposure map.

### CODE REVIEW

**Review by registerrier — APPROVED** [review]
Thanks @adonath. 

This looks good as far as I can see. 

I wonder whether a property to access `aeff` wouldn't be helpful for users.

I would rather have the exposure keyword stacking handled in the `SpectrumDataset.stack` method. It is really strange to have this specific behavior in the `RegionNDMap`.

Finally, shouldn't there be a mechanism to ensure that the `exposure` contained in the dataset is consistent with the ones in the IRF maps?

File: gammapy/datasets/spectrum.py [hunk 1]
```diff
@@ -654,7 +618,8 @@ def info_dict(self, in_safe_energy_range=True):
         mask = self.mask_safe.data if in_safe_energy_range else slice(None)
 
         info["name"] = self.name
-        info["livetime"] = self.livetime.copy()
+        if self.gti:
+            info["livetime"] = self.gti.time_sum
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/spectrum.py:622]: Yes, exactly. I think by now we don't really have unified scheme. I guess we should call this "ontime" instead?

File: gammapy/datasets/spectrum.py [hunk 2]
```diff
@@ -1288,6 +1241,11 @@ def to_ogip_files(self, outdir=None, use_sherpa=False, overwrite=False):
 
     def _ogip_meta(self):
         """Meta info for the OGIP data format"""
+        try:
+            livetime = self.exposure.meta["livetime"]
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/spectrum.py:1245]: This I don't understand. I think to be consistent with the OGIP format we have to store "livetime"?

File: gammapy/maps/core.py [hunk 3]
```diff
@@ -1134,6 +1134,32 @@ def from_images(cls, images, axis=None):
             unit=images[0].unit
         )
 
+    def to_cube(self, axes):
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/core.py:1137]: We could also just repeat the data along the axes...?

File: gammapy/maps/regionnd.py [hunk 4]
```diff
@@ -340,6 +343,13 @@ def stack(self, other, weights=None):
 
         self.data += data
 
+        # TODO: check whether this can be improved
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/regionnd.py:346]: I was 50:50 on this. I'm happy to move this to `.stack()` instead...

File: gammapy/datasets/spectrum.py [hunk 5]
```diff
@@ -506,18 +480,16 @@ def create(
         models = Models(
             [BackgroundModel(background, name=name + "-bkg", datasets_names=[name])]
         )
-        aeff = RegionNDMap.create(region=region, axes=[e_true], unit="cm2")
-        edisp = EDispKernelMap.from_diagonal_response(e_reco, e_true, counts.geom)
+        exposure = RegionNDMap.create(region=region, axes=[e_true], unit="cm2 s")
+        edisp = EDispKernelMap.from_diagonal_response(e_reco, e_true, counts.geom.to_image())
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/spectrum.py:484]: Fixed

File: gammapy/datasets/tests/test_spectrum.py [hunk 6]
```diff
@@ -91,7 +91,7 @@ def test_info_dict(spectrum_dataset):
 
     assert_allclose(info_dict["significance"], 2924.522174)
     assert_allclose(info_dict["excess"], 904010)
-    assert_allclose(info_dict["livetime"].value, 1e2)
+    assert_allclose(info_dict["livetime"].value, 216000)
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/tests/test_spectrum.py:94]: The GTI and livetime definition in the test dataset was inconsistent, see https://github.com/gammapy/gammapy/blob/master/gammapy/datasets/tests/test_spectrum.py#L60. Already this is probably a good reason, to just really on GTIs instead.

File: gammapy/datasets/spectrum.py [hunk 7]
```diff
@@ -1250,7 +1201,9 @@ def to_ogip_files(self, outdir=None, use_sherpa=False, overwrite=False):
 
         hdulist.writeto(str(outdir / phafile), overwrite=overwrite)
 
-        self.aeff.write(
+        aeff = self.exposure / self.exposure.meta["livetime"]
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/spectrum.py:1211]: We can introduce it for the `MapDataset` as well...however there is the problem of the dead time correction...

File: gammapy/maps/core.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/core.py:1137]: This I'll address in a follow up PR...

File: gammapy/maps/regionnd.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/regionnd.py]: I was wondering whether the `GTI.table.meta` is still the better place for this. Then we would just add a `GTI.livetime` helper attribute and handle it correctly in `GTI.stack()`

[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/regionnd.py]: The existence of the keyword would be fully optional...

**Comment by adonath** [comment]
@registerrier I'll go ahead and merge this, but we need a follow up PR to unify the time handling and add adding effective area properties...

**Pull request merged**